bundles/org.openhab.binding.adorne/src/main/java/org/openhab/binding/adorne/internal/discovery/AdorneDiscoveryService.java

```diff
@@ -0,0 +1,115 @@
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.adorne.internal.discovery;
+
+import static org.openhab.binding.adorne.internal.AdorneBindingConstants.*;
+
+import java.util.Collections;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.util.UIDUtils;
+import org.openhab.binding.adorne.internal.configuration.AdorneHubConfiguration;
+import org.openhab.binding.adorne.internal.hub.AdorneHubChangeNotify;
+import org.openhab.binding.adorne.internal.hub.AdorneHubController;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link AdorneDiscoveryService} discovers things for the Adorne hub and Adorne devices.
+ * Discovery is only supported if the hub is accessible via default host and port.
+ *
+ * @author Mark Theiding - Initial Contribution
+ */
+@NonNullByDefault
+@Component(service = DiscoveryService.class, immediate = true, configurationPid = "discovery.adorne")
+public class AdorneDiscoveryService extends AbstractDiscoveryService implements AdorneHubChangeNotify {
+
+    private final Logger logger = LoggerFactory.getLogger(AdorneDiscoveryService.class);
+    private static final int DISCOVERY_TIMEOUT_SECONDS = 5;
+    private static final String DISCOVERY_HUB_LABEL = "Adorne Hub";
+    private static final String DISCOVERY_ZONE_ID = "zoneId";
+
+    /**
+     * Creates a AdorneDiscoveryService with disabled auto-discovery.
+     */
+    public AdorneDiscoveryService() {
+        // Passing false as last argument to super constructor turns off background discovery
+        super(Collections.singleton(new ThingTypeUID(BINDING_ID, "-")), DISCOVERY_TIMEOUT_SECONDS, false);
+    }
+
+    /**
+     * Kick off discovery of all devices on the hub
+     */
+    @Override
+    protected void startScan() {
+        logger.debug("Discovery scan started");
+
+        // We create the hub controller with default host and port. In the future we could let users create hubs
+        // manually with custom host and port settings and then perform discovery here for those hubs.
+        AdorneHubConfiguration config = new AdorneHubConfiguration(); // Use default configuration
+        AdorneHubController adorneHubController = new AdorneHubController(config, scheduler, this);
+
+        // Limit life of hub controller to the duration of discovery
+        adorneHubController.stopBy(System.currentTimeMillis() + DISCOVERY_TIMEOUT_SECONDS * 1000);
```

Comment:
stopScan is automatically called when `DISCOVERY_TIMEOUT_SECONDS` elapses. So why not utilize that instead of using a stopBy time?

Reply:
I didn't know stopScan existed. Much simpler to leverage that. Done.
I'll propose an update to the binding documentation separately.